lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -912,28 +914,29 @@ sub part_of_important_build {
     return grep { $_ eq $build } @{$self->group->important_builds};
 }
 
-# gru job
+# obsolete gru job, replaced by JobGroups::limit_results_and_logs
 sub reduce_result {
-    my ($app, $args) = @_;
+    return;
+}
 
-    if (!ref($args)) {
-        $args = {resultdir => $args};
-    }
+sub delete_logs {
+    my ($self) = @_;
 
-    if ($args->{jobid}) {
-        my $job = $app->db->resultset('Jobs')->find({id => $args->{jobid}});
-        if ($job && $job->part_of_important_build) {
-            $app->log->debug('Job ' . $job->id . ' is part of important build, skip cleanup');
-            return;
-        }
+    if ($self->part_of_important_build) {
```

Comment:
where else are you calling this function? As this is (again) a very expensive call I would just leave that decision to the caller